backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/IrModuleSerializer.kt

```diff
@@ -0,0 +1,1003 @@
+            .setIsReified(parameter.isReified)
+        parameter.superTypes.forEach {
+            proto.addSuperType(serializeIrType(it))
+        }
+        return proto.build()
+    }
+
+    private fun serializeIrTypeParameterContainer(typeParameters: List<IrTypeParameter>): KonanIr.IrTypeParameterContainer {
+        val proto = KonanIr.IrTypeParameterContainer.newBuilder()
+        typeParameters.forEach {
+            proto.addTypeParameter(serializeDeclaration(it))
+        }
+        return proto.build()
+    }
+
+    private fun serializeIrFunctionBase(function: IrFunctionBase): KonanIr.IrFunctionBase {
+        val proto = KonanIr.IrFunctionBase.newBuilder()
+            .setName(function.name.toString())
+            .setVisibility(serializeVisibility(function.visibility))
+            .setIsInline(function.isInline)
+            .setIsExternal(function.isExternal)
+            .setReturnType(serializeIrType(function.returnType))
+            .setTypeParameters(serializeIrTypeParameterContainer(function.typeParameters))
+
+        function.dispatchReceiverParameter?.let { proto.setDispatchReceiver(serializeDeclaration(it)) }
+        function.extensionReceiverParameter?.let { proto.setExtensionReceiver(serializeDeclaration(it)) }
+        function.valueParameters.forEach {
+            //proto.addValueParameter(serializeIrValueParameter(it))
+            proto.addValueParameter(serializeDeclaration(it))
+        }
+        function.body?.let { proto.body = serializeStatement(it) }
+        return proto.build()
+    }
+
+    private fun serializeModality(modality: Modality) = when (modality) {
+        Modality.FINAL -> KonanIr.ModalityKind.FINAL_MODALITY
+        Modality.SEALED -> KonanIr.ModalityKind.SEALED_MODALITY
+        Modality.OPEN -> KonanIr.ModalityKind.OPEN_MODALITY
+        Modality.ABSTRACT -> KonanIr.ModalityKind.ABSTRACT_MODALITY
+    }
+
+    private fun serializeIrConstructor(declaration: IrConstructor): KonanIr.IrConstructor =
+        KonanIr.IrConstructor.newBuilder()
+            .setSymbol(serializeIrSymbol(declaration.symbol))
+            .setBase(serializeIrFunctionBase(declaration as IrFunctionBase))
+            .setIsPrimary(declaration.isPrimary)
+            .build()
+
+    private fun serializeIrFunction(declaration: IrSimpleFunction): KonanIr.IrFunction {
+        val function = declaration// as IrFunctionImpl
+        val proto = KonanIr.IrFunction.newBuilder()
+            .setSymbol(serializeIrSymbol(function.symbol))
+            .setModality(serializeModality(function.modality))
+            .setIsTailrec(function.isTailrec)
+            .setIsSuspend(function.isSuspend)
+
+        function.overriddenSymbols.forEach {
+            proto.addOverridden(serializeIrSymbol(it))
+        }
+
+        function.correspondingProperty?.let {
+            val uniqId = declarationTable.uniqIdByDeclaration(it)
+            proto.setCorrespondingProperty(protoUniqId(uniqId))
```

Comment:
Is it used in deserializer?

Reply:
looks like a leftover. I've removed it from konanir.proto and other places.